File: tricore-server/src/routes/master_data.rs
use actix_web::web;
use crate::handlers::master_data as md;

pub fn configure(cfg: &mut web::ServiceConfig) {
    cfg.service(
        web::scope("/master-data")
            .route("/customers", web::get().to(md::list_customers))
            .route("/customers/{id}", web::get().to(md::get_customer))
            .route("/customers", web::post().to(md::create_customer))
            .route("/customers/{id}", web::put().to(md::update_customer))
            .route("/customers/{id}", web::delete().to(md::delete_customer))
            .route("/departments", web::get().to(md::list_departments))
            .route("/departments/{id}", web::get().to(md::get_department))
            .route("/departments", web::post().to(md::create_department))
            .route("/departments/{id}", web::put().to(md::update_department))
            .route("/departments/{id}", web::delete().to(md::delete_department))
            .route("/positions", web::get().to(md::list_positions))
            .route("/positions/{id}", web::get().to(md::get_position))
            .route("/positions", web::post().to(md::create_position))
            .route("/positions/{id}", web::put().to(md::update_position))
            .route("/positions/{id}", web::delete().to(md::delete_position))
            .route("/vehicles", web::get().to(md::list_vehicles))
            .route("/vehicles/{id}", web::get().to(md::get_vehicle))
            .route("/vehicles", web::post().to(md::create_vehicle))
            .route("/vehicles/{id}", web::put().to(md::update_vehicle))
            .route("/vehicles/{id}", web::delete().to(md::delete_vehicle)),
    );
}

File: archived/tricore-web-rust/src/services/master-data.ts
import api from './http'

export const masterDataAPI = {
  // Customers
  listCustomers: () => api.get('/master-data/customers'),
  getCustomer: (id: string) => api.get(`/master-data/customers/${id}`),
  createCustomer: (data: Record<string, unknown>) => api.post('/master-data/customers', data),
  updateCustomer: (id: string, data: Record<string, unknown>) => api.put(`/master-data/customers/${id}`, data),
  deleteCustomer: (id: string) => api.delete(`/master-data/customers/${id}`),
  // Departments
  listDepartments: () => api.get('/master-data/departments'),
  getDepartment: (id: string) => api.get(`/master-data/departments/${id}`),
  createDepartment: (data: Record<string, unknown>) => api.post('/master-data/departments', data),
  updateDepartment: (id: string, data: Record<string, unknown>) => api.put(`/master-data/departments/${id}`, data),
  deleteDepartment: (id: string) => api.delete(`/master-data/departments/${id}`),
  // Positions
  listPositions: () => api.get('/master-data/positions'),
  getPosition: (id: string) => api.get(`/master-data/positions/${id}`),
  createPosition: (data: Record<string, unknown>) => api.post('/master-data/positions', data),
  updatePosition: (id: string, data: Record<string, unknown>) => api.put(`/master-data/positions/${id}`, data),
  deletePosition: (id: string) => api.delete(`/master-data/positions/${id}`),
  // Vehicles
  listVehicles: () => api.get('/master-data/vehicles'),
  getVehicle: (id: string) => api.get(`/master-data/vehicles/${id}`),
  createVehicle: (data: Record<string, unknown>) => api.post('/master-data/vehicles', data),
  updateVehicle: (id: string, data: Record<string, unknown>) => api.put(`/master-data/vehicles/${id}`, data),
  deleteVehicle: (id: string) => api.delete(`/master-data/vehicles/${id}`),
}
